### PR TITLE
refactor: order details confirm modal, fileuploader refactoring

### PR DESCRIPTION
**Pull request by nada-deriv**
## Changes:

order details confirm modal and fileuploader component =>  refactoring + typescript migration

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import React from 'react';
 import { fireEvent, render, screen, waitFor } from '@testing-library/react';
-import { useStores } from 'Stores';
-import OrderDetailsConfirmModal from '../order-details-confirm-modal.jsx';
+import { StoreProvider, mockStore } from '@deriv/stores';
 import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { useStores } from 'Stores';
```
Comment (shaheer-deriv) [inline packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx:5]: just a doubt, import should be from '@deriv/stores'?
here and in the component?

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx [hunk 2]
```diff
@@ -113,4 +106,4 @@ const OrderDetailsConfirmModal = () => {
     );
 };
 
-export default OrderDetailsConfirmModal;
+export default observer(OrderDetailsConfirmModal);
```
Comment (shaheer-deriv) [inline packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx:108]: could you please write observer at the beginning?
`const OrderDetailsConfirmModal = observer(() => {`
and keep it
`export default OrderDetailsConfirmModal;`

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx [hunk 3]
```diff
@@ -1,38 +1,36 @@
...
-    const removeFile = () => {
-        setDocumentFile({ files: [], error_message: null });
-    };
-
-    const handleRejectedFiles = files => {
-        setDocumentFile({ files, error_message: getErrorMessage(files) });
-    };
+    const {
+        ui: { is_mobile },
+    } = useStore();
```
Comment (shaheer-deriv) [inline packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx]: shall we write it as 
```
const { ui } = useStore();
const { is_mobile } = ui;
```

File: packages/p2p/src/components/file-uploader-component/file-uploader-component.tsx [hunk 4]
```diff
@@ -1,32 +1,64 @@
...
-    onDropRejected,
+    setDocumentFile,
     upload_message,
     validation_error_message,
     value,
-}) => {
+}: TFileUploaderComponentProps) => {
+    const {
+        ui: { is_mobile },
+    } = useStore();
```
Comment (shaheer-deriv) [inline packages/p2p/src/components/file-uploader-component/file-uploader-component.tsx]: here too?

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx (hunk 1 above)
Comment (nada-deriv) [reply packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx:5]: no. its a different store. p2p specific.

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx (hunk 2 above)
Comment (nada-deriv) [reply packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx:108]: hm.. i think the other way is better.. and in p2p we follow the other way round.. might be different for different packages.

Comment (shaheer-deriv) [reply packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx:108]: ok.. didn't knew things are different in p2p 😄

File: packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+    it('should upload supported file', async () => {
+        const new_props = {
+            ...props,
+            value: [file],
+        };
+        render(<FileUploaderComponent {...new_props} />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
+        const input: HTMLInputElement = screen.getByTestId('dt_file_upload_input');
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:40]: Can we use getbyrole query here? it seems like it's an input

File: packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    });
+
+    it('should show error message when unsupported file is uploaded', async () => {
+        const new_props = { ...props, validation_error_message: 'error' };
+        render(<FileUploaderComponent {...new_props} />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
+
+        const unsupported_file = new File(['hello'], 'hello.html', { type: 'html' });
+        const input = screen.getByTestId('dt_file_upload_input');
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:57]: Can we getthisbyrole?

Comment (niloofar-deriv) [reply packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:57]: image

File: packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            expect(screen.getByText('error')).toBeInTheDocument();
+        });
+    });
+    it('should handle remove File', async () => {
+        const new_props = { ...props, validation_error_message: 'error' };
+        render(<FileUploaderComponent {...new_props} />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
+
+        const input: HTMLInputElement = screen.getByTestId('dt_file_upload_input');
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:70]: Same for this

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx [hunk 8]
```diff
@@ -78,31 +70,35 @@ describe('<OrderDetailsConfirmModal/>', () => {
...
         const { confirmOrderRequest, order_information } = order_store;
 
-        render(<OrderDetailsConfirmModal />);
+        render(<OrderDetailsConfirmModal />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
         const file = new File(['hello'], 'hello.png', { type: 'image/png' });
-        const input = screen.getByTestId('dt_file_upload_input');
-        fireEvent.change(input, { target: { files: [file] } });
+        const input: HTMLInputElement = screen.getByTestId('dt_file_upload_input');
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/__tests__/order-details-confirm-modal.spec.tsx:91]: same

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx (hunk 3 above)
Comment (ameerul-deriv) [reply packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx]: hmmm i think the above is perfectly fine, it shouldn't cause any errors IMO and there are other files where we have destructured it in that way before

File: packages/p2p/src/components/file-uploader-component/file-uploader-component.tsx (hunk 4 above)
Comment (ameerul-deriv) [reply packages/p2p/src/components/file-uploader-component/file-uploader-component.tsx]: ditto here

File: packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx [hunk 9]
```diff
@@ -1,38 +1,36 @@
 import React from 'react';
 import { useStores } from 'Stores';
 import { Button, Modal, Text } from '@deriv/components';
-import { isMobile, formatMoney } from '@deriv/shared';
+import { formatMoney, TFile } from '@deriv/shared';
+import { observer, useStore } from '@deriv/stores';
 import { localize } from '@deriv/translations';
 import FileUploaderComponent from 'Components/file-uploader-component';
+//Todo: change the below line when changing the base branch
```
Comment (ameerul-deriv) [inline packages/p2p/src/components/modal-manager/modals/order-details-confirm-modal/order-details-confirm-modal.tsx]: `nit`: i think we can be more specific here with this todo, and just say when merging with our refactor reusable components branch

File: packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx (hunk 5 above)
Comment (nada-deriv) [reply packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:40]: checked this. the role is not accessible.

File: packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx (hunk 6 above)
Comment (nada-deriv) [reply packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx:57]: as discussed, same as above.

**Review by ameerul-deriv — APPROVED** [review]
LGTM

**Review by niloofar-deriv — APPROVED** [review]

**Review by farhan-nurzi-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**